x-pack/plugin/security/src/test/java/org/elasticsearch/xpack/security/authc/jwt/JwtStringClaimValidatorTests.java

```diff
@@ -167,6 +180,253 @@ public void testDoesNotSupportWildcardOrRegex() throws ParseException {
         }
     }
 
```

Comment:
Nice job on testing. 
small nit:  We could potentially add tests which demonstrate that: 
1. `sub` validation is case sensitive
2. `?` can be used as a single special character that must be present
3.  an empty (`""`) `sub` is rejected

Reply:
Additional test suggestion, we could adjust existing [JwtRestIT](https://github.com/elastic/elasticsearch/blob/fcdeb95652e4e1703746ed0f85138a948aa23298/x-pack/plugin/security/qa/jwt-realm/src/javaRestTest/java/org/elasticsearch/xpack/security/authc/jwt/JwtRestIT.java#L152C1-L152C1) integration test to randomly include `allowed_subjects` or `allowed_subject_patterns` (e.g. `service_*@app?.example.com`).